build.sh

```diff
@@ -23,11 +23,58 @@ fi
 
 __scriptpath=$(cd "$(dirname "$0")"; pwd -P)
 
-"$__scriptpath/build-native.sh" $*
+__ServerGC=0
+__IsCentos=0
+__IsOSX=0
+
+while :; do
+    if [ $# -le 0 ]; then
+        break
+    fi
+    
+    if [[ $1 == *"centos.7"* ]] ; then
+       __IsCentos=1
+    fi
+    
+    if [[ $1 == *"osx.10"* ]] ; then
+       __IsOSX=1
+    fi
+
+    lowerI="$(echo $1 | awk '{print tolower($0)}')"
+    case $lowerI in
+        useservergc)
+            __ServerGC=1
+            ;;
+        *)
+    __UnprocessedBuildArgs="$__UnprocessedBuildArgs $1"
+    esac
+
+    shift
+done
+
+export CORECLR_SERVER_GC="$__ServerGC"
+
+"$__scriptpath/build-native.sh" $__UnprocessedBuildArgs
 if [ $? -ne 0 ];then
    exit 1
 fi
 
-"$__scriptpath/build-managed.sh" $*
-exit $?
 
+if [ $__IsCentos -eq 1 ] ; then
```

Comment:
Can you please file follow-up issues for removing the logic from this sh script, both the servergc logic and the os detection.